test_conformance/vulkan/test_vulkan_interop_buffer.cpp

```diff
@@ -1688,108 +1545,99 @@ int test_buffer_common(cl_device_id device_, cl_context context_,
     uint32_t bufferSizeListforOffset[] = { 256, 512, 1024 };
 
     cl_context_properties contextProperties[] = { CL_CONTEXT_PLATFORM, 0, 0 };
+    std::vector<VulkanExternalSemaphoreHandleType> supportedSemaphoreTypes;
+
     errNum = clGetPlatformIDs(1, &platform, NULL);
-    if (errNum != CL_SUCCESS)
-    {
-        print_error(errNum, "Error: Failed to get platform\n");
-        goto CLEANUP;
-    }
+    test_error_and_cleanup(errNum, CLEANUP, "Error: Failed to get platform\n");
 
     errNum =
         clGetDeviceIDs(platform, CL_DEVICE_TYPE_GPU, 0, NULL, &num_devices);
-    if (CL_SUCCESS != errNum)
-    {
-        print_error(errNum, "clGetDeviceIDs failed in returning of devices\n");
-        goto CLEANUP;
-    }
+    test_error_and_cleanup(errNum, CLEANUP,
+                           "clGetDeviceIDs failed in returning of devices\n");
+
     devices = (cl_device_id *)malloc(num_devices * sizeof(cl_device_id));
     if (NULL == devices)
     {
-        errNum = CL_OUT_OF_HOST_MEMORY;
-        print_error(errNum, "Unable to allocate memory for devices\n");
-        goto CLEANUP;
+        test_fail_and_cleanup(errNum, CLEANUP,
+                              "Unable to allocate memory for devices\n");
     }
     errNum = clGetDeviceIDs(platform, CL_DEVICE_TYPE_GPU, num_devices, devices,
                             NULL);
-    if (CL_SUCCESS != errNum)
-    {
-        print_error(errNum, "Failed to get deviceID.\n");
-        goto CLEANUP;
-    }
+    test_error_and_cleanup(errNum, CLEANUP, "Failed to get deviceID.\n");
+
     contextProperties[1] = (cl_context_properties)platform;
     log_info("Assigned contextproperties for platform\n");
     for (device_no = 0; device_no < num_devices; device_no++)
     {
-        errNum = clGetDeviceInfo(devices[device_no], CL_DEVICE_EXTENSIONS, 0,
-                                 NULL, &extensionSize);
-        if (CL_SUCCESS != errNum)
+        errNum = clGetDeviceInfo(devices[device_no], CL_DEVICE_UUID_KHR,
+                                 CL_UUID_SIZE_KHR, uuid, NULL);
+        test_error_and_cleanup(errNum, CLEANUP, "clGetDeviceInfo failed\n");
+
+        if (!is_extension_available(devices[device_no],
+                                    "cl_khr_external_memory_opaque_fd"))
         {
-            print_error(errNum,
-                        "Error in clGetDeviceInfo for getting device_extension "
-                        "size....\n");
-            goto CLEANUP;
+            log_info(
+                "Device %u does not support cl_khr_external_memory_opaque_fd, "
+                "which is required for this test. Skipping\n",
+                device_no);
+            continue;
```

Comment:
What if cl_khr_external_memory_opaque_fd is not present, but some other handle type extension is present (e.g. cl_khr_external_memory_win32)? Consider either removing the check or adding checks for other handle types too.

Reply:
Done